modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/baseComponents/BaseAPISchemaFields.tsx

```diff
@@ -37,8 +38,16 @@ export default function BaseAPISchemaFields({
 			filter: 'status/any(k:k eq 0)',
 			url: '/o/object-admin/v1.0/object-definitions',
 		}).then((result) => {
-			const options = result
-				? result.map((objectDefinition) => ({
+			const filteredResult = result.filter(
+				(option) =>
+					!option.system ||
```

Comment:
Instad of `system`, you can use modifiable == true and you can eliminate all the API builder definitions dependency

Reply:
I will continue reviewing the PR